spring-pulsar/src/main/java/org/springframework/pulsar/core/DefaultPulsarConsumerFactory.java

```diff
@@ -38,14 +42,15 @@
  * @param <T> underlying payload type for the consumer.
  * @author Soby Chacko
  * @author Alexander Preuß
+ * @author Christophe Bornet
  */
 public class DefaultPulsarConsumerFactory<T> implements PulsarConsumerFactory<T> {
 
 	private final Map<String, Object> consumerConfig = new HashMap<>();
 
 	private final List<Consumer<T>> consumers = new ArrayList<>();
 
-	private PulsarClient pulsarClient;
+	private final PulsarClient pulsarClient;
 
 	public DefaultPulsarConsumerFactory(PulsarClient pulsarClient, Map<String, Object> consumerConfig) {
 		this.pulsarClient = pulsarClient;
```

Comment:
Wdyt about making the default config map immutable? I think it would be a good idea. Also, we can advertise this fact in `PCF.getConsumerConfig`.

Reply:
done via `Collections.unmodifiable`.
Or do you consider another immutable impl ?

Reply:
Thats what I would have used as well. 👍🏻